### PR TITLE
Eventual send

### PR DESCRIPTION
**Pull request by michaelfig**
This PR adds infix bang/eventual send support.

~It requires ses-0.5.1, which is not yet released, so for now, it just uses a tarball to install it.  When SES is released, the tarball requirement will be removed from the commit.~

Please discuss the other changes.

### CODE REVIEW

File: src/kernel/liveSlots.js [hunk 1]
```diff
@@ -17,9 +17,31 @@ function build(syscall, _state, makeRoot, forVatID) {
...
+    // Make a handler that uses the E proxy.
+    const handler = {
+      GET(o, prop) {
+        return E(o)[prop];
+      },
+      PUT(o, prop, value) {
+        return (E(o)[prop] = value);
+      },
+      POST(o, prop, args) {
+        return prop === undefined ? E(o)(...args) : E(o)[prop](...args);
```
Comment (erights) [inline src/kernel/liveSlots.js]: Yeah, I think you should flip this around. This handler should do the work that E would have done, but without calling E.

**Comment by michaelfig** [comment]
Note that 15 of the test cases still fail: apparently, the ordering and how many messages are sent per step of the kernel has changed.  Looking at the runQueue gives different results than before handled Promises were introduced.

I'm not sure if that's because our test cases are overspecified, or if it's a real bug.

**Comment by michaelfig** [comment]
The tests now pass!  This was [Agoric/eventual-send#18](https://github.com/Agoric/eventual-send/issues/18).

@warner, could you please do a review, then merge and release?  As soon as we have a release, I'll verify that @agoric/cosmic-swingset also works with it.

**Comment by warner** [comment]
Huh, I did a quick `npm test` on my machine at home, and got test failures. I find it hard to know exactly which test is failing (grumble grumble `tape` displays descriptions not testfile names grumble), but it involves `basedir-controller-3` and some of the output is:

```
    doProcess: vat[_bootstrap][0].bootstrap dispatch failed: TypeError: Promise.makeHandled is not a function TypeError: Promise.makeHandled is not a function
        at makeQueued (eval at <anonymous> (eval at createScopedEvaluatorFactory (/home/warner/stuff/agoric/SwingSet/node_modules/ses/dist/ses.cjs.js:2688:10)), <anonymous>:609:20)
        at unserializeSlot (eval at <anonymous> (eval at createScopedEvaluatorFactory (/home/warner/stuff/agoric/SwingSet/node_modules/ses/dist/ses.cjs.js:2688:10)), <anonymous>:737:20)
        at fullRevive (eval at <anonymous> (eval at createScopedEvaluatorFactory (/home/warner/stuff/agoric/SwingSet/node_modules/ses/dist/ses
```

Any idea why this is behaving differently than CI? I tried both Nodejs v12.3.1 and v12.6.0, they appeared to behave the same.

BTW I landed an earlier PR, so if you make significant changes to this branch, feel free to rebase. But once it looks good I'm happen to do a merge without rebasing.

I haven't studied the code at all yet, but I will.

**Comment by warner** [comment]
BTW I've got SES-0.5.1 in my `node_modules/` (we released 0.5.2 just today, but I don't think the changes are relevant)

**Comment by michaelfig** [comment]
Please run `find . -name .cache -print0 | xargs -0 rm -rf && npm install`.  You'll probably notice that the test runs differently after that.

ESM's module cache is easily confused by node_modules updates.

**Comment by warner** [comment]
Indeed, that did the trick, thanks. Reviewing now.

**Comment by michaelfig** [comment]
I rebased just now.

**Comment by warner** [comment]
Oh and I'm guessing #85 can be closed once this lands.

**Comment by michaelfig** [comment]
I'll include the TC39 pointer.

Actually, non-SES rollup does work with infix bang.  Just not `eval` within the rollup bundles (if that's even possible).

**Comment by warner** [comment]
Ah, right, because then `eval` is coming from the local `@agoric/evaluate` module we install (whose source lives in SwingSet, in `./agoric_evaluate/`). I wonder if it's possible to modify that to know about the rewriting? I guess the use case is dynamic bang-using contract code that gets evaluated at runtime by the Contract Host, when run in SwingSet in `--no-ses` mode.

**Comment by warner** [comment]
BTW, we merged the "E: implement in terms of handled Promises" commit (18d28319483d78b20c02504846ce688c6d37cdfd) on trunk and made a new 0.0.15 release (4c89956875d8223b6d60d6854a77add885e522b8) to include it. The README changes will probably go into a new PR, and this one will be closed.

**Comment by michaelfig** [comment]
The rest of this change is in #92

**Pull request closed without merging**